packages/esm-patient-registration-app/src/patient-registration/field/name/name-field.component.tsx

```diff
@@ -19,17 +19,18 @@ function checkNumber(value: string) {
 }
 
 export const NameField = () => {
-  const {
-    fieldConfigurations: {
-      name: { displayCapturePhoto },
-    },
-  } = useConfig() as RegistrationConfig;
+  const { fieldConfigurations } = useConfig<RegistrationConfig>();
   const { t } = useTranslation();
-  const { setCapturePhotoProps, currentPhoto, setFieldValue } = useContext(PatientRegistrationContext);
-  const { fieldConfigurations } = useConfig();
+  const { setCapturePhotoProps, currentPhoto, setFieldValue, initialFormValues } =
+    useContext(PatientRegistrationContext);
   const fieldConfigs = fieldConfigurations?.name;
   const [{ value: unidentified }] = useField('unidentifiedPatient');
   const nameKnown = !unidentified;
+  const showToggleUnknownName =
+    fieldConfigs?.unidentifiedPatient ||
+    (initialFormValues?.familyName === 'UNKNOWN' && initialFormValues?.givenName === 'UNKNOWN')
+      ? true
+      : false;
```

Comment:
Since the `nameKnown` constant is used for toggling yes or no, it couldn't be used to show the toggle when editing an UNKNOWN patient. So, I had to directly use the initial form values to track if a patient is unknown or not.

This works fine for the name field since we have a way to track if a patient is known or not. But for the date of birth, I think there is hardly any way to identify whether this D.O.B was estimated or not.

Reply:
@Jexsie - both the "unknown name" and "estimated date of birth" data points are available in the patient record.  

For "unknown name", you can get an idea of how this is stored if you look at "form-manager.ts", where you can see this block of code:

```
    if (values.unidentifiedPatient) {
      attributes.push({
        // The UUID of the 'Unknown Patient' attribute-type will always be static across all implementations of OpenMRS
        attributeType: '8b56eac7-5c76-4b9c-8c6f-1deab8d3fc47',
        value: 'true',
      });
    }
```

So, you will be able to determine if an existing patient has "unknown name" = true if they have an attribute with attributeType.uuid = `8b56eac7-5c76-4b9c-8c6f-1deab8d3fc47` and value = `true`.  This is what you need to look for, not the fact that the name fields are set as UNKNOWN, UNKNOWN.

Estimated date of birth is even more explicit in the data model.  The `patient` and `person` resources themselves have a `birthdateEstimated` property that you can check.  See the `Patient` interface defined in `patientOperations.ts` for example.

Please use these fields, which is what is actually set in the data in the back-end when these are entered in the registration form.

Reply:
Thanks @mseaton for the help.